src/CaptureService/CaptureService.cpp

```diff
@@ -116,7 +116,7 @@ void CaptureService::FinalizeEventProcessing(StopCaptureReason stop_capture_reas
   producer_event_processor_->ProcessEvent(orbit_grpc_protos::kRootProducerId,
                                           std::move(capture_finished));
 
-  grpc_client_capture_event_collector_->StopAndWait();
+  grpc_client_capture_event_collector_->OnStopAndWaitRequestedByExternal();
```

Comment:
It seems that's the only call of `StopAndWait`. Can you quickly explain your plans? Is this code going to be refactored into a separate function?

Reply:
My current plan is:
- Change `CaptureService::StartEventProcessing` and `CaptureService::FinalizeEventProcessing` to be free functions which can be used without a service, 
```cpp
uint64_t StartEventProcessing(
    const orbit_grpc_protos::CaptureOptions& capture_options,
    orbit_producer_event_processor::ProducerEventProcessor* producer_event_processor,
    uint64_t clock_resolution_ns);

void FinalizeEventProcessing(
    StopCaptureReason stop_capture_reason,
    orbit_producer_event_processor::ProducerEventProcessor* producer_event_processor,
    orbit_producer_event_processor::ClientCaptureEventCollector* client_capture_event_collector);
```
which support `ClientCaptureEventProcessor` rather than `GrpcClientCaptureEventProcessor`.
These two methods are used in `LinuxCaptureService::Capture`, which is the method we would like to use without a actual service and reader_writer.

Reply:
Ah, that makes sense. Thank you for the explanation!